freeipa/src/main/java/com/sequenceiq/freeipa/service/proxy/ModifyProxyConfigOrchestratorService.java

```diff
@@ -0,0 +1,57 @@
+package com.sequenceiq.freeipa.service.proxy;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.orchestrator.exception.CloudbreakOrchestratorException;
+import com.sequenceiq.cloudbreak.orchestrator.host.HostOrchestrator;
+import com.sequenceiq.cloudbreak.orchestrator.host.OrchestratorStateParams;
+import com.sequenceiq.freeipa.entity.InstanceMetaData;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.orchestrator.OrchestratorStateParamsProvider;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+@Service
+public class ModifyProxyConfigOrchestratorService {
+
+    static final String MODIFY_PROXY_STATE = "modifyproxy";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ModifyProxyConfigOrchestratorService.class);
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private OrchestratorStateParamsProvider orchestratorStateParamsProvider;
+
+    @Inject
+    private HostOrchestrator hostOrchestrator;
+
+    public void applyModifyProxyState(Long stackId) throws CloudbreakOrchestratorException {
+        Stack stack = stackService.getByIdWithListsInTransaction(stackId);
+        List<InstanceMetaData> sortedInstances = getSortedInstances(stack);
+
+        for (InstanceMetaData instance : sortedInstances) {
+            String hostName = instance.getDiscoveryFQDN();
+            OrchestratorStateParams stateParams = orchestratorStateParamsProvider.createStateParamsForSingleTarget(stack, hostName, MODIFY_PROXY_STATE);
+            LOGGER.debug("Calling applyModifyProxyState for instance {} with state params '{}'", hostName, stateParams);
+            hostOrchestrator.runOrchestratorState(stateParams);
+        }
```

Comment:
just to have a note here too: @Bajzathd will add connectivity check here, and also ensure that proxy modification will update primary gateway at the end, after the others are successfully finished